docs/modeling-your-data/modeling-your-data-with-dbt/dbt-normalize-model/index.md

```diff
@@ -0,0 +1,298 @@
+
+  - `users_table` _(macro)_: This macro takes a series of inputs to generate the SQL that will produce your users table, using the `user_id` column and any custom contexts from your events table.
+
+  - `snowplow_normalize_model_gen.py` _(script)_: This script uses an input configuration to generate the models based on a provided configuration file. See the [operation](#operation) section for more information.
+
+    ```yml title="snowplow_normalize_model_gen.py help"
+    usage: snowplow_normalize_model_gen.py [-h] [--version] [-v] [--dryRun] [--configHelp] [--cleanUp] config
+
+    Produce dbt model files for normalizing your Snowplow events table into 1 table per event type.
+
+    positional arguments:
+    config         relative path to your configuration file
+
+    optional arguments:
+    -h, --help     show this help message and exit
+    --version      show program's version number and exit
+    -v, --verbose  verbose flag for the running of the tool
+    --dryRun       flag for a dry run (does not write/delete any files)
+    --configHelp   prints information relating to the structure of the config file
+    --cleanUp      delete any models not present in your config and exit (no models will be generated)
+    ```
+
+  - `example_normalize_config.json`: This file is an example of an input to the python script, showing all options and valid values. For additional information about the file structure run `python dbt_packages/snowplow_normalize/utils/snowplow_normalize_model_gen.py --configHelp` in your project root.
+
+  - `example_resolver_config.json`: This file is an example [Iglu Resolver](/docs/pipeline-components-and-applications/iglu/iglu-resolver/index.md) configuration. It supports custom iglu servers with API keys, but does not currently support accessing embedded registries. For more information please see the Resolver docs.
+
+## Operation
+
+In general, it should only be required to run the script in this package once to begin with, then only as/when you need to add new models based on new events or alter the contexts attached to existing ones. While it is possible to manually set the values and use the macros, it is not recommended due to the time it would take and the likelihood of making mistakes.
+
+:::caution
+The script should always be run from the **root** of your dbt project (the same level your `dbt_project.yml` file is at).
+:::
+
+### Install python packages
+:::caution
+Python [>=3.7, <=3.10] is currently supported
+
+:::
+
+The script only requires 2 additional packages (`jsonschema` and `requests`) that are not built into python by default, you can install these by running the below command, or by installing them by your preferred method.
+
+```bash
+pip install dbt_packages/snowplow_normalize/utils/requirements.txt
```

Comment:
I believe a the -r flag is missing here, at least it only worked for me like that. 
```bash 
pip install -r dbt_packages/snowplow_normalize/utils/requirements.txt
```